presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -136,19 +139,42 @@ private DistributedQueryRunner(
             discoveryServer = new TestingDiscoveryServer(environment);
             closer.register(() -> closeUnchecked(discoveryServer));
             log.info("Created TestingDiscoveryServer in %s", nanosSince(start).convertToMostSuccinctTimeUnit());
+            URI discoveryUrl = discoveryServer.getBaseUrl();
+            log.info("Discovery URL %s", discoveryUrl);
+
+            if (nativeWorkerLauncher.isPresent()) {
+                ImmutableList.Builder<Process> nativeWorkersBuilder = ImmutableList.builder();
+                for (int i = 0; i < nodeCount; i++) {
+                    nativeWorkersBuilder.add(nativeWorkerLauncher.get().apply(i, discoveryUrl));
+                }
+                nativeWorkers = nativeWorkersBuilder.build();
+                closer.register(() -> {
+                    for (Process nativeWorker : nativeWorkers) {
+                        nativeWorker.destroyForcibly();
+                    }
+                });
+            }
+            else {
+                nativeWorkers = ImmutableList.of();
+            }
 
             ImmutableList.Builder<TestingPrestoServer> servers = ImmutableList.builder();
-
-            for (int i = 1; i < nodeCount; i++) {
-                TestingPrestoServer worker = closer.register(createTestingPrestoServer(discoveryServer.getBaseUrl(), false, extraProperties, parserOptions, environment, baseDataDir));
-                servers.add(worker);
+            if (!nativeWorkerLauncher.isPresent()) {
+                for (int i = 1; i < nodeCount; i++) {
+                    TestingPrestoServer worker = closer.register(createTestingPrestoServer(discoveryUrl, false, extraProperties, parserOptions, environment, baseDataDir));
+                    servers.add(worker);
+                }
             }
 
             Map<String, String> extraCoordinatorProperties = new HashMap<>();
+            if (nativeWorkerLauncher.isPresent()) {
+                // Don't use coordinator as worker
+                extraCoordinatorProperties.put("node-scheduler.include-coordinator", "false");
+            }
```

Comment:
maybe just fail it hard and force the caller to set `"node-scheduler.include-coordinator", "false"`?

Reply:
@highker Not sure I understand the suggestion. Would you like me to assert that coordinatorProperties include "node-scheduler.include-coordinator" with the value of "false"?